### PR TITLE
AntennaAttributionDocumentReporter: Aggregate copyright statements

### LINKED ISSUE #3200: Make AntennaAttributionDocumentReporter used "merged" / processed copyrights
The `AntennaAttributionDocumentReporter` currently still report individual Copyright statements like

```
Copyright (c) 2001-2011 Michael Niedermayer <michaelni@gmx.at>
Copyright (c) 2002 Michael Niedermayer <michaelni@gmx.at>
Copyright (c) 2003 Michael Niedermayer <michaelni@gmx.at>
Copyright (c) 2006 Michael Niedermayer <michaelni@gmx.at>
```

although these should have been merged to a single `Copyright (c) 2001-2011 Michael Niedermayer <michaelni@gmx.at>` entry. We probably miss to (implicitly) make use of the `CopyrightStatementsProcessor` for some reason. Looking at other reporters like `NoticeTemplateReporter` might give a clue what is going on.

### PR DESCRIPTION
**Pull request by MarcelBochtler**
Merge copyright entries and filter excluded copyrights.

Resolves #3200.

### CODE REVIEW

File: reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt [hunk 1]
```diff
@@ -238,12 +239,16 @@ class AntennaAttributionDocumentReporter : Reporter {
         return outputFiles
     }
 
-    private fun createCopyrightStatement(resolvedLicense: ResolvedLicenseInfo) =
-        resolvedLicense.flatMapTo(sortedSetOf()) { license ->
+    private fun createCopyrightStatement(resolvedLicense: ResolvedLicenseInfo): String {
```
Comment (fviernau) [inline reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt:242]: The `Reporter` also gets (optionally) a `CopyrightGarbage` list as input.
How about applying the garbage filter before handing the statements over to the `CopyrightStatementsProcessor` ?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt (hunk 1 above)
Comment (mnonnenmacher) [reply reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt:242]: As the code uses the `resolvedLicense` copyright garbage was already filtered.

File: reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt [hunk 2]
```diff
@@ -238,12 +239,16 @@ class AntennaAttributionDocumentReporter : Reporter {
...
+        val copyrights = resolvedLicense.filterExcluded().flatMapTo(sortedSetOf()) { license ->
             license.locations.flatMap { location ->
                 location.copyrights.map { it.statement }
             }
-        }.joinToString("\n")
+        }
```
Comment (mnonnenmacher) [inline reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt:247]: I would recommend to replace `license.locations.flatMap { ...}` with `license.getCopyrights(omitExcluded = true)`.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt (hunk 2 above)
Comment (MarcelBochtler) [reply reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt]: Using `license.getCopyrights()` which also omits the excluded copyrights, therefore I removed the `filterExcluded()`

**Review by fviernau — APPROVED** [review]

**Pull request merged**